### PR TITLE
[#2313] Porting IVC+ADD 6: Add folding expression evaluation

### PR DESCRIPTION
**Pull request by volhovm**

### CODE REVIEW

File: ivc/tests/add.rs [hunk 1]
```diff
@@ -341,7 +346,7 @@ pub fn heavy_test_simple_add() {
         }
     }
 
-    type Config = StandardConfig<
+    type Config<const N_COL: usize, const N_FSEL: usize> = StandardConfig<
```
Comment (dannywillems) [inline ivc/tests/add.rs:349]: Why this change?

Comment (dannywillems) [reply ivc/tests/add.rs:349]: It is adding noise, but ok... I would simply remove this.

File: ivc/tests/add.rs [hunk 2]
```diff
@@ -416,13 +562,13 @@ pub fn heavy_test_simple_add() {
     );
 
     // Make the constraints folding compatible
-    let app_compat_constraints: Vec<FoldingCompatibleExpr<Config>> = app_constraints
+    let app_compat_constraints: Vec<FoldingCompatibleExpr<MainTestConfig>> = app_constraints
```
Comment (dannywillems) [inline ivc/tests/add.rs:565]: Noise from the change above. The PR should be only the changes above.

File: ivc/tests/add.rs [hunk 3]
```diff
@@ -350,6 +355,147 @@ pub fn heavy_test_simple_add() {
         (),
         GenericVecStructure<Curve>,
     >;
+    type MainTestConfig = Config<N_COL_TOTAL, N_FSEL_TOTAL>;
+
+    ////////////////////////////////////////////////////////////////////////////
+    // Evaluations
+    ////////////////////////////////////////////////////////////////////////////
+
+    #[allow(dead_code)]
```
Comment (dannywillems) [inline ivc/tests/add.rs:364]: I would guess it is used later.

File: ivc/tests/add.rs (hunk 1 above)
Comment (volhovm) [reply ivc/tests/add.rs:349]: It will be generalized and moved out of this file (into the library) 5 PRs later, to be used with different circuits.

Here it was originally introduced because I was experimenting with several setups at the same time.

**Review by dannywillems — APPROVED** [review]
Approving as other PRs are built on top of this, but I would still remove the noise added by the type alias.

File: ivc/tests/add.rs (hunk 3 above)
Comment (volhovm) [reply ivc/tests/add.rs:364]: Is removed eventually as far as I remember.
